components/eamxx/src/share/field/field_utils.hpp

```diff
@@ -179,6 +179,99 @@ void horiz_contraction(const Field &f_out, const Field &f_in,
   impl::horiz_contraction<ST>(f_out, f_in, weight, comm);
 }
 
+// Utility to compute the contraction of a field along its level dimension.
+// This is equivalent to f_out = einsum('...k->...', weight, f_in).
+// The impl is such that:
+// - f_out, f_in, and weight must be provided and allocated
+// - The last dimension is for the levels (LEV)
+// - There can be only up to 3 dimensions of f_in
+// - Weight is assumed to be (in order of checking/impl):
+//   - rank-1, with only LEV dimension
+//   - rank-2, with only COL and LEV dimensions
+template <typename ST>
+void vert_contraction(const Field &f_out, const Field &f_in,
+                      const Field &weight, const ekat::Comm *comm = nullptr) {
+  using namespace ShortFieldTagsNames;
+
+  const auto &l_out = f_out.get_header().get_identifier().get_layout();
+
+  const auto &l_in = f_in.get_header().get_identifier().get_layout();
+
+  const auto &l_w = weight.get_header().get_identifier().get_layout();
+
+  // Sanity checks before handing off to the implementation
+  EKAT_REQUIRE_MSG(
+      l_w.rank() >= 1 && l_w.rank() <= 2,
```

Comment:
Probably subjective, but since there are only 2 options, we may do a more explicit check like `l_w.rank()==1 or l_w.rank()==2`, which has the same code length anyways. Inequalities take a bit longer to visually parse, and it may take some time for the reader to figure out that 1 and 2 are the only valid values.